sourceHandlers/bitBucket.js

```diff
@@ -0,0 +1,170 @@
+const SourceData = require('../model/sourceData');
+const async = require('async');
+const request = require('request');
+const SourceHandler = require('./sourceHandler');
+
+const userAgent = 'SoftwareCitationCore';
+
+
+/*
+ * Creates and sends a request for the BitBucket API.
+ * @private
+ * @param {string} path - The path of the url.  This does not include the base path.  For example if you
+ * want to send a request to 'http://api.github.com/repos/apple/swift' use 'repos/apple/swift'.
+ * @param {Function} cb - The callback function. Follows the error/response parameter pattern.
+ * The response param will be json parsed object.
+ */
+function sendApiRequest(url, path, cb) {
+  const options = {
+    url: url + path,
+    headers: {
+      'User-Agent': userAgent,
+    },
+  };
+
+  request(options, (error, response, body) => {
+    const parsedBody = (body != null) ? JSON.parse(body) : null;
+    if (error == null && response.statusCode !== 200) {
+      const errorMessage = `Received a ${response.statusCode} when making a request to ${options.url}`;
+      cb(new Error(errorMessage), parsedBody);
+    }
+    else {
+      cb(error, parsedBody);
+    }
+  });
+}
+
+/*
+ * Quries the the BitBucket API to get the authors of a project
+ * @param {string[]} gitHubLogins - An array of BitBucket username to query for information.
+ * @param {Function} callback - The callback function. Follows the error response parameter pattern.
+ * The response parameter is an array of Author objects
+ */
+function getAllCommits(url, path, obj, cb) {
```

Comment:
Name does not match functionality described in comment. Params do not match those described in comment.